app/src/org/commcare/connect/PersonalIdManager.java

```diff
@@ -342,6 +342,7 @@ private void showLinkDialog(CommCareActivity<?> activity, ConnectLinkedAppRecord
         dialog.setNegativeButton(activity.getString(R.string.login_link_connectid_no), (d, w) -> {
             activity.dismissAlertDialog();
             ConnectAppDatabaseUtil.storeApp(activity, linkedApp);
+            FirebaseAnalyticsUtil.reportPersonalIDLinking(linkedApp.getAppId(),false);
```

Comment:
we should also send the info on whether the user pushed negative button to cause a failure here or it was a connect unlock failure, As such I suggest that we send a string valued parameter with these values instead of a boolean - `success`, `failure_user_denied`, `failure_unlock_failed`